.buildkite/winlogbeat/pipeline.winlogbeat.yml

```diff
@@ -2,48 +2,149 @@
-  PACKAGING_ARM_PLATFORMS: "linux/arm64"
-  PACKAGING_PLATFORMS: "+all linux/amd64 linux/arm64 windows/amd64 darwin/amd64 darwin/arm64"
-
-  #Deps
+  # Other deps
   ASDF_MAGE_VERSION: 1.15.0
 
 steps:
+  - group: "Winlogbeat Mandatory Tests"
+    key: "winlogbeat-mandatory-tests"
+
+    steps:
+      - label: ":ubuntu: Winlogbeat Crossccompile"
+        key: "mandatory-cross-compile"
+        command: "make -C winlogbeat crosscompile"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_UBUNTU_X86_64}"
+          machineType: "${GCP_DEFAULT_MACHINE_TYPE}"
+        artifact_paths:
+          - "winlogbeat/build/*.xml"
+          - "winlogbeat/build/*.json"
+        notify:
+          - github_commit_status:
+              context: "Winlogbeat: Crosscompile"
+
+      - label: ":windows: Winlogbeat Win-2016 Unit Tests"
+        command: |
+          Set-Location -Path winlogbeat
+          mage build unitTest
+        key: "mandatory-win-2016-unit-tests"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_WIN_2016}"
+          machine_type: "${GCP_WIN_MACHINE_TYPE}"
+          disk_size: 100
+          disk_type: "pd-ssd"
+        artifact_paths:
+          - "winlogbeat/build/*.xml"
+          - "winlogbeat/build/*.json"
+        notify:
+          - github_commit_status:
+              context: "Winlogbeat: Win-2016 Unit Tests"
+
+      - label: ":windows: Winlogbeat Win-2019 Unit Tests"
+        command: |
+          Set-Location -Path winlogbeat
+          mage build unitTest
+        key: "mandatory-win-2019-unit-tests"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_WIN_2019}"
+          machineType: "${GCP_WIN_MACHINE_TYPE}"
+          disk_size: 100
+          disk_type: "pd-ssd"
+        artifact_paths:
+          - "winlogbeat/build/*.xml"
+          - "winlogbeat/build/*.json"
+        notify:
+          - github_commit_status:
+              context: "Winlogbeat: Win-2019 Unit Tests"
 
-  - input: "Input Parameters"
-    key: "input-run-all-stages"
-    fields:
-    - select: "Winlogbeat - run_winlogbeat"
-      key: "run_winlogbeat"
-      options:
-        - label: "True"
-          value: "true"
-        - label: "False"
-          value: "false"
-      default: "false"
-    if: "build.source == 'ui'"
+      - label: ":windows: Winlogbeat Win-2022 Unit Tests"
+        command: |
+          Set-Location -Path winlogbeat
+          mage build unitTest
+        key: "mandatory-win-2022-unit-tests"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_WIN_2022}"
+          machine_type: "${GCP_WIN_MACHINE_TYPE}"
+          disk_size: 100
+          disk_type: "pd-ssd"
+        artifact_paths:
+          - "winlogbeat/build/*.xml"
+          - "winlogbeat/build/*.json"
+        notify:
+          - github_commit_status:
+              context: "Winlogbeat: Win-2022 Unit Tests"
+
+  - group: "Extended Windows Tests"
+    key: "winlogbeat-extended-win-tests"
+
```

Comment:
should we add:

`if: build.env("BUILDKITE_PULL_REQUEST") == "false" || build.env("GITHUB_PR_LABELS") =~ /.*[Ww]indows.*/`


to allow running with the right label this extended group (see https://github.com/elastic/beats/blob/main/.buildkite/metricbeat/pipeline.yml#L158C5-L158C108)